Bottom: 59cee822dde81991f1226b8f6d4688620f4bec05
Top:    bd4c7b19caf1d28aabe4411466c968abf767fef1
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-04-20 00:08:10 -0400

Refresh of efika-stac9766

---

diff --git a/sound/soc/fsl/soc-of-simple.c b/sound/soc/fsl/soc-of-simple.c
index 49c07f6..2b32f64 100644
--- a/sound/soc/fsl/soc-of-simple.c
+++ b/sound/soc/fsl/soc-of-simple.c
@@ -131,8 +131,7 @@ int of_snd_soc_register_codec(struct snd_soc_codec_device *codec_dev,
 	of_soc->device.codec_dev = codec_dev;
 	of_soc->card.num_links = min(count, of_soc->card.num_links);
 	for (i = 0; i < of_soc->card.num_links; i++) {
-		of_soc->dai_link[i].name = (char *)node->name;
-		of_soc->dai_link[i].stream_name = (char *)node->name;
+		of_soc->dai_link[i].name = dai[i].name;
 		of_soc->dai_link[i].codec_dai = dai;
 	}
 	/* Now try to register the SoC device */
@@ -202,8 +201,10 @@ int of_snd_soc_register_cpu_dai(struct device_node *node,
 
 	of_soc->cpu_dai_node = node;
 	of_soc->card.num_links = min(count, of_soc->card.num_links);
-	for (i = 0; i < of_soc->card.num_links; i++)
+	for (i = 0; i < of_soc->card.num_links; i++) {
+		of_soc->dai_link[i].stream_name = cpu_dai[i].name;
 		of_soc->dai_link[i].cpu_dai = &cpu_dai[i];
+	}
 
 	/* Now try to register the SoC device */
 	of_snd_soc_register_device(of_soc);
